### PR TITLE
OCPCLOUD-2638: Add stricter linting configuration

### PR DESCRIPTION
**Pull request by JoelSpeed**
This updates the linting in this repo to match largely with what we have done in other places, like CPMS.

The linting is now quite strict and we may find we want to remove some of these, but I think for now, lets try this out and see how we get on.

Ended up going through and fixing all the linting issues, so in theory, everything should now be compliant 👀

### CODE REVIEW

File: pkg/controllers/capiinstaller/watch_predicates.go [hunk 1]
```diff
@@ -54,7 +67,10 @@ func ownedPlatformLabelPredicate(namespace string, platform configv1.PlatformTyp
 
 // isOwnedProviderComponent checks whether an object is an owned provider component.
 func isOwnedProviderComponent(obj runtime.Object, namespace string, platform configv1.PlatformType) bool {
-	cO := obj.(client.Object)
+	cO, ok := obj.(client.Object)
```
Comment (theobarberbany) [inline pkg/controllers/capiinstaller/watch_predicates.go:70]: nice catch from the linter

File: pkg/controllers/infracluster/infracluster_controller.go [hunk 2]
```diff
@@ -1,7 +1,23 @@
+/*
```
[the remark this answers is not in the document]
Comment (theobarberbany) [reply pkg/controllers/infracluster/infracluster_controller.go:1]: Do we want to turn this linter off?

[the remark this answers is not in the document]
Comment (JoelSpeed) [reply pkg/controllers/infracluster/infracluster_controller.go:1]: This linter, whatever it was, was an accident 😅 It's something built into prow that I accidentally triggered

**Review by theobarberbany — COMMENTED** [review]
/lgtm

**Review by damdo — COMMENTED** [review]
Awesome, thanks!

/approve

**Comment by JoelSpeed** [comment]
/label acknowledge-critical-fixes-only

This has no effect on the code itself, but I want to merge it ahead of other PRs else it will need rework

**Pull request merged**